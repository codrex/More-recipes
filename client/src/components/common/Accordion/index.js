export { default as Accordion } from './Accordion';
export { default as AccordionHeader } from './AccordionHeader';
export { default as AccordionBody } from './AccordionBody';
