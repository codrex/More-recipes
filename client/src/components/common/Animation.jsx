import { CSSTransitionGroup } from 'react-transition-group';
import React from 'react';
import PropTypes from 'prop-types';

const Animation = props => (
  <CSSTransitionGroup
    transitionName={props.name}
    transitionAppearTimeout={props.timeOut}
    transitionLeaveTimeout={props.timeOut}
    transitionEnterTimeout={props.timeOut}
    transitionAppear={props.appear}
  >
    {props.children}
  </CSSTransitionGroup>
);

Animation.propTypes = {
  children: PropTypes.node.isRequired,
  name: PropTypes.string,
  timeOut: PropTypes.number,
  appear: PropTypes.bool
};

Animation.defaultProps = {
  name: 'fade',
  timeOut: 600,
  appear: true
};

export default Animation;
