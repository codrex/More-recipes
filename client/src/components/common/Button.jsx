import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

/**
 * Button component
 * @param {object} props
 * @return {React} react element
 */
const Button = (props) => {
  const {
    Tag,
    handleClick,
    dataTarget,
    dataToggle,
    href,
    type,
    id,
  } = props;
  return (
    <Tag
      type={type}
      className={classnames('btn', props.className)}
      data-dismiss="modal"
      data-toggle={dataToggle}
      data-target={dataTarget}
      onClick={handleClick}
      href={href}
      id={id}
    >
      {props.text}
      {props.children}
    </Tag>
  );
};

Button.propTypes = {
  type: PropTypes.string,
  text: PropTypes.string,
  className: PropTypes.string,
  dataToggle: PropTypes.string,
  dataTarget: PropTypes.string,
  handleClick: PropTypes.func,
  children: PropTypes.node,
  Tag: PropTypes.string,
  href: PropTypes.string,
  id: PropTypes.string
};
Button.defaultProps = {
  type: 'button',
  text: '',
  className: '',
  dataToggle: '',
  dataTarget: '',
  handleClick: () => {},
  children: null,
  Tag: 'button',
  href: '',
  id: ''
};

export default Button;
