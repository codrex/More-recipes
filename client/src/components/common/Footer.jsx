import React from 'react';

/**
 * Footer component
 * @param {object} props
 * @return {React} react element
 */
const Footer = () => (
  <footer className="footer w-100">
    <div className="row">
    Â© 2017 Copyright Andela.com
    </div>
  </footer>
);

export default Footer;
