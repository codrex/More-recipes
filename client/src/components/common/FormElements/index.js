export { default as Form } from './Form';
export { default as Input } from './Input';
export { default as PureInput } from './PureInput';
export { default as Select } from './Select';
export { default as Textarea } from './Textarea';
export { default as PureTextarea } from './PureTextarea';
