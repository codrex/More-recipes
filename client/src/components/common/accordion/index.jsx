export { default as Accordion } from './accordion';
export { default as AccordionHeader } from './accordionHeader';
export { default as AccordionBody } from './accordionBody';
