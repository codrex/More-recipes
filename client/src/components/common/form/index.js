export { default as Form } from './form';
export { default as Input } from './input';
export { default as PureInput } from './pureInput';
export { default as Select } from './select';
export { default as Textarea } from './textarea';
export { default as PureTextarea } from './pureTextarea';
