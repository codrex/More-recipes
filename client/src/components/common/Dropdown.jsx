import React from 'react';
import PropTypes from 'prop-types';
import Animation from './Animation';


/**
 * Dropdown component
 * @param {object} props
 * @return {React} react component
 */
class Dropdown extends React.Component {
  /**
   * @return {undefined}
   * @param {object} props
   */
  constructor(props) {
    super(props);
    this.state = {
      isOpen: false
    };
  }

  /**
   * @return {undefined}
   */
  componentWillReceiveProps() {
    if (this.state.isOpen) {
      this.setState({ isOpen: false });
    }
  }

  /**
   * hides the dropdown menu
   * @return {undefined}
   */
  hideDropdown = () => {
    this.setState({ isOpen: false });
  }

  /**
   * display the dropdown menu
   * @return {undefined}
   */
  showDropdown = () => {
    this.setState({ isOpen: true });
  }

  /**
   * @return {React} List
   * render dropdown menu list
   */
  renderList = () => {
    if (!this.state.isOpen) return null;
    const {
      options,
      headerText
    } = this.props;

    return (
      <Animation name="dropdown-menu" timeOut={300}>
        <ul className="dropdown-menu">
          {headerText && <li
            className="dropdown-item dropdown-header"
            key={headerText}
          >
            {headerText}
          </li>}
          {headerText && <hr style={{ marginTop: 0 }} key="hrHeader" />}
          {
            options.map((option) => {
              const { label, action } = option;
              if (label === 'separator') {
                return <hr style={{ marginBottom: 0 }} key="separator" />;
              }
              return (
                <li
                  key={label}
                  className="dropdown-item"
                >
                  <div
                    onClick={action}
                    role="button"
                    tabIndex="0"
                  >
                    {label}
                  </div>
                </li>
              );
            })
          }
        </ul>
      </Animation>
    );
  }

  /**
   * render an overlay component
   * @return {React} Overlay
   */
  renderOverlay = () => {
    if (this.state.isOpen) {
      return (
        <div
          className="dropdown-overlay"
          onClick={this.hideDropdown}
          role="button"
          tabIndex="0"
        />
      );
    }
    return null;
  }

  /**
   * renders dropdown trigger
   * @return {React} trigger
   */
  renderTrigger = () => {
    const { Trigger } = this.props;
    if (Trigger) {
      return (
        <Trigger
          className="dropdown-toggle"
          onClick={this.showDropdown}
        />
      );
    }
    return (
      <div
        className="dropdown-toggle"
        onClick={this.showDropdown}
        role="button"
        tabIndex="0"
      >
        drop down
      </div>
    );
  }

  /**
   * dropdown
   * @return {React} dropdown
   */
  render() {
    return (
      <div
        className="dropdown"
      >
        {this.renderOverlay()}
        {this.renderTrigger()}
        {this.renderList()}
      </div>
    );
  }
}

Dropdown.propTypes = {
  Trigger: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
  options: PropTypes.arrayOf(PropTypes.object).isRequired,
  headerText: PropTypes.string
};

Dropdown.defaultProps = {
  Trigger: null,
  headerText: null
};

export default Dropdown;
