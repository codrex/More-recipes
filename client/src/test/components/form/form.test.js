/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import toJson from 'enzyme-to-json';
import { Form } from '../../../components/common/FormElements';

const props = {
  pageCount: 5,
  handlePageClick: jest.fn(),
  loading: false
};

describe('Form component ', () => {
  test('expected to match snapshot', () => {
    const component = mount(<Form {...props} />);
    const tree = toJson(component);
    component.find('form').simulate('submit');
    expect(tree).toMatchSnapshot();
  });
});
