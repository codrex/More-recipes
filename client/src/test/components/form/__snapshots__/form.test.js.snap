// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component  expected to match snapshot 1`] = `
<Form
  action=""
  className=""
  disabled={false}
  handlePageClick={[Function]}
  id="submit"
  lg={true}
  loading={false}
  method=""
  onSubmit={[Function]}
  pageCount={5}
  primary={false}
  primaryInverse={false}
  secondary={false}
  secondaryInverse={false}
  style={Object {}}
  submitBtnText=""
>
  <form
    action=""
    className="col-12"
    method=""
    onSubmit={[Function]}
    style={Object {}}
  />
</Form>
`;
